Libplanet/Blockchain/BlockChain.cs

```diff
@@ -17,13 +17,23 @@ namespace Libplanet.Blockchain
     public class BlockChain<T> : IEnumerable<Block<T>>
         where T : IAction
     {
+        private readonly Guid _id;
+
         public BlockChain(IBlockPolicy<T> policy, IStore store)
+            : this(Guid.NewGuid(), policy, store)
```

Comment:
As the `id` parameter is optional in users' perspective, shouldn't it be the last parameter?

Reply:
User can see only `BlockChain(policy, store)` because `BlockChain(id, policy, store)` is private. 
but I also think that moving `id` to last as you say is better. I'll change.